lib/elixir/lib/system.ex

```diff
@@ -759,6 +759,11 @@ defmodule System do
   `System.convert_time_unit(1, :second, :native)`).
   """
   @spec convert_time_unit(integer, time_unit | :native, time_unit | :native) :: integer
+  # 0 will always convert to 0
+  def convert_time_unit(0, _, _) do
+    0
+  end
```

Comment:
The issue with this is that we won't check the other arguments are valid. And then if we have to check if the other arguments are valid, then I am afraid we won't see any gain in performance, so I would probably not merge this. :) Thoughts?

Reply:
I see. So it is consistent and raises on invalid units? Would it make sense to do this pattern matching in a private function in the DateTime module that does not check for the other arguments? For instance in add we can call it for the offsets, but call the System module directly for  `ppd = System.convert_time_unit(86400, :second, unit)` since that is not going to be 0 and then it will raise is the unit is not valid.

Reply:
Doing it inside calendar is fine by ultimately I don't think it will matter as this is supposed to be quite fast anyway. :) Your call.